src/vector/backends/object.py

```diff
@@ -662,17 +681,29 @@ def from_rhophi(cls, rho: float, phi: float) -> "VectorObject2D":
             >>> import vector
             >>> vec = vector.VectorObject2D.from_rhophi(1, 2)
             >>> vec
-            vector.obj(rho=1, phi=2)
+            VectorObject2D(rho=1, phi=2)
         """
-        return cls(AzimuthalObjectRhoPhi(rho, phi))
+        return cls(azimuthal=AzimuthalObjectRhoPhi(rho, phi))
 
-    def __init__(self, azimuthal: AzimuthalObject) -> None:
-        self.azimuthal = azimuthal
+    def __init__(
+        self, azimuthal: typing.Optional[AzimuthalObject] = None, **kwargs: float
+    ) -> None:
+        if not kwargs and azimuthal is not None:
+            self.azimuthal = azimuthal
+        elif kwargs and azimuthal is None:
+            if set(kwargs) == {"x", "y"}:
+                self.azimuthal = AzimuthalObjectXY(kwargs["x"], kwargs["y"])
+            elif set(kwargs) == {"rho", "phi"}:
+                self.azimuthal = AzimuthalObjectRhoPhi(kwargs["rho"], kwargs["phi"])
+            else:
+                raise TypeError("invalid arguments, must be x=, y= or rho=, phi=")
+        else:
+            raise TypeError("must give Azimuthal if not giving keyword arguments")
```

Comment:
Removing the `MomentumObject` constructor and changing the `momentum` coordinates to `generic` ones works!
```suggestion
    def __init__(
        self, azimuthal: typing.Optional[AzimuthalObject] = None, **kwargs: float
    ) -> None:
        for k, v in kwargs.copy().items():
            kwargs.pop(k)
            kwargs[_repr_momentum_to_generic.get(k, k)] = v

        if not kwargs and azimuthal is not None:
            self.azimuthal = azimuthal
        elif kwargs and azimuthal is None:
            if set(kwargs) == {"x", "y"}:
                self.azimuthal = AzimuthalObjectXY(kwargs["x"], kwargs["y"])
            elif set(kwargs) == {"rho", "phi"}:
                self.azimuthal = AzimuthalObjectRhoPhi(kwargs["rho"], kwargs["phi"])
            else:
                raise TypeError("invalid arguments, must be x=, y= or rho=, phi=")
        else:
            raise TypeError("must give Azimuthal if not giving keyword arguments")
```